lib/commands/context.js

```diff
@@ -0,0 +1,461 @@
+    if (ctx.view && (ctx.view.title || '').match(titleRegex)) {
+      if (ctx.view.url === 'about:blank') {
+        // in the case of Xcode  < 5 (i.e., iOS SDK Version less than 7)
+        // and in the case of iOS 7.1 in a webview (not in Safari)
+        // we can have the url be `about:blank`
+        if (parseFloat(this.iOSSDKVersion) < 7 ||
+            (this.caps.platformVersion === '7.1' && this.caps.app && this.caps.app.toLowerCase() !== 'safari')) {
+          matchingCtx = ctx;
+        }
+      } else {
+        matchingCtx = ctx;
+      }
+    }
+  }
+  return matchingCtx ? matchingCtx.id : undefined;
+};
+
+// Right now we don't necessarily wait for webview
+// and frame to load, which leads to race conditions and flakiness,
+// let's see if we can transition to something better
+extensions.useNewSafari = function () {
+  return parseFloat(this.iosSdkVersion) >= 8.1 &&
+         parseFloat(this.caps.platformVersion) >= 8.1 &&
+         !this.isRealDevice() &&
+         this.caps.safari;
+};
+
+extensions.navToInitialWebview = async function () {
+  let timeout = 0;
+  if (this.isRealDevice()) {
+    timeout = 3000;
+    logger.debug(`Waiting for ${timeout} ms before navigating to view.`);
+  }
+  await B.delay(timeout);
+  if (this.useNewSafari()) {
+    await this.typeAndNavToUrl();
+  } else if (parseInt(this.iosSdkVersion, 10) >= 7 && !this.isRealDevice() && this.caps.safari) {
+    await this.navToViewThroughFavorites();
+  } else {
+    await this.navToViewWithTitle(/.*/);
+  }
+};
+
+extensions.typeAndNavToUrl = async function () {
+  let initialUrl = this.caps.safariInitialUrl || `http://127.0.0.1:${this.caps.port}/welcome`;
+  let oldImpWait = this.implicitWaitMs;
+  this.implicitWaitMs = 7000;
+  let el = await this.findElement('name', 'URL');
+  this.implicitWaitMs = oldImpWait;
+  await this.nativeTap(el.ELEMENT);
+  let els = await this.findElements('name', 'Address');
+  let addressEl = els[els.length - 1].ELEMENT;
+  await this.setValueImmediate(addressEl, initialUrl);
+  el = await this.findElement('name', 'go');
+  await this.nativeTap(res.ELEMENT);
+  await this.navToViewWithTitle(/.*/i);
+  // Waits for page to finish loading.
+  await this.remote.pageUnload();
+};
+
+extensions.navToViewThroughFavorites = async function () {
+  logger.debug('We are on iOS7+ simulator: clicking apple button to get into a webview');
+  let oldImpWait = this.implicitWaitMs;
+  this.implicitWaitMs = 7000; // wait 7s for apple button to exist
+  try {
+    let el = await this.findElement('xpath', '//UIAScrollView[1]/UIAButton[1]');
+  } catch (err) {
+    let msg = 'Could not find button to click to get into webview. ' +
+              'Proceeding on the assumption we have a working one.';
+    logger.error(msg);
+    this.implicitWaitMs = oldImpWait;
+    return await this.navToViewWithTitle(/.*/i);
+  }
+  this.implicitWaitMs = oldImpWait;
+  try {
+    let res = await this.nativeTap(el.ELEMENT);
+    if (res.status !== 0) {
```

Comment:
is this another erroneous assumption about receiving a jsonwp response instead of an actual value? can we not get rid of this try/catch and simply allow `nativeTap` to throw if it doesn't work?